crates/core/src/fnc/rand.rs

```diff
@@ -125,7 +125,26 @@ pub fn string((arg1, arg2): (Option<i64>, Option<i64>)) -> Result<Value, Error>
 	Ok(Alphanumeric.sample_string(&mut rand::thread_rng(), val).into())
 }
 
-pub fn time((range,): (Option<(i64, i64)>,)) -> Result<Value, Error> {
+pub fn time((range,): (Option<(Value, Value)>,)) -> Result<Value, Error> {
+	// Process the arguments
+	let range =
+		match range {
+			None => None,
+			Some((Value::Number(Number::Int(min)), Value::Number(Number::Int(max)))) => {
+				Some((min, max))
+			}
+			Some((Value::Datetime(min), Value::Datetime(max))) => {
+				let min = min.to_i64();
```

Comment:
In addition to [making `to_i64` fallible](https://github.com/surrealdb/surrealdb/pull/5202/files#r1862617957), I think we should return an error when either `min` or `max` can't be converted to `i64` instead of falling back to `0`.

Reply:
Yeah agree with @rushmorem here 👍

Reply:
Fixed in [464a9b1](https://github.com/surrealdb/surrealdb/pull/5202/commits/464a9b1ba2fa3bd7b3af2069152983e40acb834d)